src/pandablocks_ioc/_tables.py

```diff
@@ -274,35 +291,47 @@ def __init__(
             value,
         )
 
-        for (field_name, field_record_container), data in zip(
-            self.table_fields_records.items(), field_data
-        ):
+        putorder_index = 0
+
+        for field_name, field_record_container in self.table_fields_records.items():
             field_details = field_record_container.field
 
             full_name = table_name + ":" + field_name
             full_name = EpicsName(full_name)
             description = trim_description(field_details.description, full_name)
 
+            waveform_val = self._construct_waveform_val(
+                field_data, field_name, field_details
+            )
+
             field_record: RecordWrapper = builder.WaveformOut(
                 full_name,
                 DESC=description,
                 validate=self.validate_waveform,
                 on_update_name=self.update_waveform,
-                initial_value=[str(x).encode() for x in data],
+                initial_value=waveform_val,
                 length=field_info.max_length,
             )
+
+            pva_info = {
+                f"value.{field_name}": {
```

Comment:
I think the PVA table should have lowercase field names by convention, so this should probably be
```suggestion
                f"value.{field_name.lower()}": {
```